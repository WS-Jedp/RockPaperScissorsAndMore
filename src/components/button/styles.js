import styled from 'styled-components'

export const ButtonSecondaryWrapper = styled.button`
  display: flex;
  position: relative;
  align-items: center;
  justify-content: center;
  min-width: 130px;
  min-height: 42px;
  border: 2px solid var(--white-color);
  border-radius: 12px;
  color: var(--white-color);
  font-size: 15px;
  font-weight: 500;
  letter-spacing: 2px;
  background-color: transparent;
  text-transform: uppercase;
  cursor: pointer;
  transition: 0.3s ease-in-out;

  &:hover {
    box-shadow: 0 0 12px 3px rgba(255,255,255, .3)
  }


`

export const ButtonPrimaryWrapper = styled.button`
  display: flex;
  position: relative;
  align-items: center;
  justify-content: center;
  min-width: 190px;
  min-height: 42px;
  border: 2px solid var(--white-color);
  border-radius: 12px;
  color: var(--black-color);
  font-size: 15px;
  font-weight: 700;
  letter-spacing: 2px;
  background-color: var(--white-color);
  text-transform: uppercase;
  cursor: pointer;
  transition: 0.3s ease-in-out;

  &:hover {
    box-shadow: 0 0 12px 3px rgba(255,255,255, .3)
  }


`