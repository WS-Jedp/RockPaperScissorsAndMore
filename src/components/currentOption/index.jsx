import React from 'react'
import { IconLizard, IconPaper, IconScissors, IconRock, IconSpock } from '../icons'

export const CurrentOption = ({option = 'rock'}) => {

  const optionName = option.toLowerCase(); 

    switch (optionName) {
      case "rock":
        return <IconRock />
        break;
      case "paper":
        return <IconPaper />
        break;
      case "scissors":
    return <IconScissors />
        break;
      case "lizard":
        return <IconLizard />
        break;
      case "spock":
        return <IconSpock />
        break;
    
      default:
        return <IconRock />
        break;
    }
  }