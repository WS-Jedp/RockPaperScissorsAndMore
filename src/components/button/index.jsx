import React from 'react'
import { ButtonSecondaryWrapper, ButtonPrimaryWrapper } from './styles'

export const ButtonSecondary = ({title = 'Rules', action = () => {}}) => (
  <ButtonSecondaryWrapper onClick={action}>
    {title}
  </ButtonSecondaryWrapper>
)

export const ButtonPrimary = ({title = 'Play Again', action = () => {}}) => (
  <ButtonPrimaryWrapper onClick={action}>
    {title}
  </ButtonPrimaryWrapper>
)
