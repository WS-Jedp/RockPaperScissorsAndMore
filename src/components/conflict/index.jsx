import React from 'react'
import { ConflictWrapper } from './styles'

import { PickedOption } from '../pickedOption'
import { PlayAgain } from '../playAgain'

export const Conflict = ({userOption = {}, homeOption = {}}) => {

  const isMobile = false
  
  return (
    <>
      <ConflictWrapper>
        <PickedOption title='You Picked' option={userOption.name} color={userOption.color} picked={true} />
        {
        isMobile && <PlayAgain title='You Win'action={() => {}} />
        }
        <PickedOption title='The House Picked' option={homeOption.name} color={homeOption.color} />
      </ConflictWrapper>
      {
        !isMobile && <PlayAgain title='You Win'action={() => {}} />
      }
    </>
  )
}
