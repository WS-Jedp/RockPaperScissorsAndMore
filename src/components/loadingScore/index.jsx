import React from 'react'
import { LoadingWrapper } from './styles'

export const LoadingScore = () => (
  <LoadingWrapper />
)
