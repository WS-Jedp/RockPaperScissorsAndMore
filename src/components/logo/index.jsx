import React from 'react'
import logoSVG from '../../assets/images/logo-bonus.svg'

import { LogoWrapper } from './styles'

export const Logo = () => (
  <LogoWrapper aria-label="Logo Rock, Paper, Scissors And More">
    <img src={logoSVG} alt="Logo Rock, Paper, Scissors And More"/>
  </LogoWrapper>
)