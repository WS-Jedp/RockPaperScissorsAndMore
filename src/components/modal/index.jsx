import React from 'react'
import ReactDOM from 'react-dom'
import { ModalWrapper } from './styles'

export const Modal = ({children}) => ReactDOM.createPortal((
  <ModalWrapper>
    { children || 'Hello World' }
  </ModalWrapper>), 
  window.document.getElementById('modal-portal'))
