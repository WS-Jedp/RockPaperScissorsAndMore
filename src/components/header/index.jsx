import React from 'react'
import { WrapperHeader } from './styles'

import { Score } from '../score/index'
import { Logo } from '../logo'

export const Header = () => (
  <WrapperHeader>
    <Logo />
    <Score />
  </WrapperHeader>
) 