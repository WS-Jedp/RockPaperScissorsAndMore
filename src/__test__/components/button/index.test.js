import React from 'react'
import { create } from 'react-test-renderer'
import { mount } from 'enzyme'

import { ButtonPrimary, ButtonSecondary } from '../../../components/button'

describe('Tests to the Button compoennt', () => {
  describe('Visual tests to the "Button" component', () => {
    const ButtonComponentPrimary = create(<ButtonPrimary />)
    const ButtonComponentSecondary = create(<ButtonSecondary />)

    test('The components should be the same', () => {
      expect(ButtonComponentPrimary.toJSON()).toMatchSnapshot()
      expect(ButtonComponentSecondary.toJSON()).toMatchSnapshot()
    })
  })

  describe('Functional tests to the "Button" component', () => {
    test('should the title of the component be a string', () => {
      expect(typeof mount(<ButtonPrimary />).find('button').text()).toBe('string')
      expect(typeof mount(<ButtonSecondary />).find('button').text()).toBe('string')
    })
    
  })
})
