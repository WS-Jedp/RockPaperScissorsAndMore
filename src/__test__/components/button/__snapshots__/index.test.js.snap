// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests to the Button compoennt Visual tests to the "Button" component The components should be the same 1`] = `
<button
  className="sc-AxirZ eZZyOJ"
  onClick={[Function]}
>
  Play Again
</button>
`;

exports[`Tests to the Button compoennt Visual tests to the "Button" component The components should be the same 2`] = `
<button
  className="sc-AxjAm kAaWoH"
  onClick={[Function]}
>
  Rules
</button>
`;
